use-cases.md

```diff
@@ -10,19 +10,18 @@ This page aims to illustrate common use cases that developers who instrument the
 
 ```python
     def top_level_function():
-        span1 = tracer.start_trace('top_level_function')
+        ctx, span1 = tracer.start_trace('top_level_function')
         try:
             . . . # business logic
         finally:
             span1.finish()
 ```
 
-As a follow-up, suppose that as part of the business logic above we call another `function2` that we also want to trace. In order to attach that function to the ongoing trace, we need a way to access `span1`. We discuss how it can be done later, for now let's assume we have a helper function `get_current_span` for that:
+As a follow-up, suppose that as part of the business logic above we call another `function2` that we want to report on in our trace. In order to attach a Span reporting this function to the ongoing trace, we need to provide the Context that carries the trace state.. We discuss how it can be done later, for now let's assume we have a helper function `get_current_context` for that:
 
 ```python
     def function2():
-        span2 = get_current_span().start_child('function2') \
-            if get_current_span() else None
+        span2 = tracer.create_span(get_current_context(), 'function2')
```

Comment:
really no reason for both this an join_trace to exist (and the consolidated thing could reasonably be named either... I guess I'd go for `create_span`, but whatever).